tests/test_config.py

```diff
@@ -0,0 +1,141 @@
+        assert cfg.value('preload1dict1.preload2_1val3') == '2_1val3'
+        assert cfg.value('preload1dict1.main1val3') == 'main1val3'
+        assert cfg.value('preload1dict1.augment1val1') == 'a1_1val1'
+
+        assert cfg.value('preload1dict2') == '2scalar1'
+
+        assert cfg.value('preload1list1') == [ 'main1' ]
+        assert cfg.value('preload1list2') == [ '1_2val0', '1_2val1', 'a1_2val0' ]
+
+        assert cfg.value('preload1scalar1') == '2scalar1'
+        assert cfg.value('preload1scalar2') == '1scalar2'
+        assert cfg.value('preload1scalar3') == 'main3'
+        assert cfg.value('preload2scalar2') == '2scalar2'
+
+    def test_main(self, cfg):
+        assert cfg.value('maindict.mainval1') == 'val1'
+        assert cfg.value('maindict.mainval2') == 'val2'
+
+        assert cfg.value('mainlist1') == ['main1', 'main2', 'main3', 'aug1', 'aug2']
+        assert cfg.value('mainlist2') == ['override1', 'override2']
+
+        assert cfg.value('mainscalar1') == 'aug1'
+        assert cfg.value('mainscalar2') == 'override2'
+        assert cfg.value('mainscalar3') == 'override2'
+
+        assert cfg.value('mainnull') is None
+        with pytest.raises( ValueError ):
+            print( f"config value notdefined is {cfg.value('notdefined')}" )
+
+    def test_override(self, cfg):
+        assert cfg.value( 'override1list1.0' ) == '1_1override1'
+        assert cfg.value( 'override1list1.1' ) == '1_1override2'
+        assert cfg.value( 'override1list2' ) == [ '2_2override1', '2_2override2' ]
+
+    def test_fieldsep( self, cfg ):
+        fields, isleaf, curfield, ifield = cfg._fieldsep( 'nest.nest1.0.nest1a' )
+        assert isleaf == False
+        assert curfield == 'nest'
+        assert fields == ['nest', 'nest1', '0', 'nest1a' ]
+        assert ifield is None
+        fields, isleaf, curfield, ifield = cfg._fieldsep( '0.test' )
+        assert isleaf == False
+        assert ifield == 0
+        fields, isleaf, curfield, ifield = cfg._fieldsep( 'mainlist2' )
+        assert isleaf
+        fields, isleaf, curfield, ifield = cfg._fieldsep( 'mainscalar1' )
+        assert isleaf
+
+
+    def test_nest(self, cfg):
+        assert cfg.value( 'nest' ) ==  { 'nest1': [ { 'nest1a': { 'val': 'foo' } }, 42 ],
+                                         'nest2': { 'val': 'bar' } }
+        assert cfg.value( 'nest.nest1.0.nest1a.val' ) == 'foo'
+
+    def test_set(self, cfg):
+        with pytest.raises( TypeError ):
+            cfg.set_value( 'settest.list.notanumber', 'kitten', appendlists=True )
+        with pytest.raises( TypeError ):
+            cfg.set_value( 'settest.0', 'puppy' )
+        with pytest.raises( TypeError ):
+            cfg.set_value( 'settest.0.subset', 'bunny' )
+        with pytest.raises( TypeError ):
+            cfg.set_value( 'settest.dict.0', 'iguana' )
+        with pytest.raises( TypeError ):
+            cfg.set_value( 'settest.dict.2.something', 'tarantula' )
```

Comment:
Maybe add a `match=...` clause to all of the `raises` calls?

Reply:
I guess I don't understand what you're thinking here.

Reply:
OH.

I was being stupid.  I thought you  meant the python match statement; I see what you mean now.